partner_phone_extension/models/res_partner.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2014 Savoir-faire Linux
+#   (<http://www.savoirfairelinux.com>).
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    extension = fields.Char('Extension', help="Phone Number Extension.")
```

Comment:
I guess the name should be `phone_extension`, adding `oldname="extension"` to the field definition.